mmv1/products/datastream/Stream.yaml

```diff
@@ -764,6 +784,211 @@ properties:
+          - source_config.0.sql_server_source_config
+        description: |
+          SQL Server data source configuration.
+        properties:
+          - !ruby/object:Api::Type::NestedObject
+            name: 'includeObjects'
+            description: |
+              SQL Server objects to retrieve from the source.
+            properties:
+              - !ruby/object:Api::Type::Array
+                name: 'schemas'
+                required: true
+                min_size: 1
+                description: |
+                  SQL Server schemas/databases in the database server
+                item_type: !ruby/object:Api::Type::NestedObject
+                  description: |
+                    SQL Server database.
+                  properties:
+                    - !ruby/object:Api::Type::String
+                      name: 'schema'
+                      required: true
+                      description: |
+                        Schema name.
+                    - !ruby/object:Api::Type::Array
+                      name: 'tables'
+                      min_size: 1
+                      description: |
+                        Tables in the database.
+                      item_type: !ruby/object:Api::Type::NestedObject
+                        description: |
+                          SQL Server table.
+                        properties:
+                          - !ruby/object:Api::Type::String
+                            name: 'table'
+                            required: true
+                            description: |
+                              Table name.
+                          - !ruby/object:Api::Type::Array
+                            name: 'columns'
+                            min_size: 1
+                            description: |
+                              SQL Server columns in the schema. When unspecified as part of include/exclude objects, includes/excludes everything.
+                            item_type: !ruby/object:Api::Type::NestedObject
+                              description: |
+                                SQL Server Column.
+                              properties:
+                                - !ruby/object:Api::Type::String
+                                  name: 'column'
+                                  description: |
+                                    Column name.
+                                - !ruby/object:Api::Type::String
+                                  name: 'dataType'
+                                  description: |
+                                    The SQL Server data type. Full data types list can be found here:
+                                    https://learn.microsoft.com/en-us/sql/t-sql/data-types/data-types-transact-sql?view=sql-server-ver16
+                                - !ruby/object:Api::Type::Integer
+                                  name: 'length'
+                                  output: true
```

Comment:
The API docs don't seem to indicate this & the following fields are output-only; why are they marked `output: true`?

Reply:
I had followed what was in the [oracle source config](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/products/datastream/Stream.yaml#L423C33-L457C85) which had `output: true` set for those fields even though they are also not listed as output-only in the [API docs](https://cloud.google.com/datastream/docs/reference/rest/v1/projects.locations.streams#oraclecolumn)

Fixed in 97c904b9ceac7b210d337f403550cf4e1f15af05

Reply:
Any chance you can confirm that they are in fact settable? (For example by setting them in tests.)

Reply:
Sure give me a few mins

Reply:
reverted: 93f26b4683bd132046ca42f7297d0d5bdd8980c2

Reply:
```tf
Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.precision":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.primary_key":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.scale":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.length":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.nullable":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.ordinal_position":
│ its value will be decided automatically based on the result of
│ applying this configuration.
╵
```

Reply:
hm, that looks like the fields are still marked as output-only; you'll need to remove the `output: true` in order to test whether the API allows setting them.

Reply:
Sorry posted the messages in reverse order... I should have posted the revert message after the error log.

I tested using 93f26b4683bd132046ca42f7297d0d5bdd8980c2 which has the lines `output: true` removed. See below for the second test:

Commit Used to Build the Provider
```sh
git log
commit 93f26b4683bd132046ca42f7297d0d5bdd8980c2 (HEAD -> feat/datastream-sql-server)
Author: nbugden <nbugden@gmail.com>
Date:   Mon May 6 13:19:49 2024 -0400

    Revert "chore: remove beta-only"
    
    This reverts commit 939b4f1e8962c3ac939f24d2fc98170f48bdc4d7.
```

TF Config:
```tf
resource "google_datastream_stream" "default" {
  provider = google-beta
  project  = sensitive(var.project_id)

  stream_id     = "${var.prefix}-stream"
  desired_state = "NOT_STARTED"
  location      = var.region
  display_name  = "${var.prefix}-stream"

  source_config {
    source_connection_profile = google_datastream_connection_profile.source_connection_profile.id
    sql_server_source_config {
      include_objects {
        schemas {
          schema = sensitive(var.sql_server_config.schema)
          tables {
            table = sensitive(var.sql_server_config.table)
            columns {
              column           = "test"
              length           = 1
              precision        = 1
              scale            = 1
              primary_key      = true
              nullable         = true
              ordinal_position = 1
            }
          }
        }
      }
    }
  }

  destination_config {
    destination_connection_profile = google_datastream_connection_profile.destination_connection_profile.id
    gcs_destination_config {
      path = var.bucket.path
      json_file_format {
        schema_file_format = "NO_SCHEMA_FILE"
        compression        = "GZIP"
      }
    }
  }

  backfill_none {}

}
```

Output:
```tf
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.precision":
│ its value will be decided automatically based on the
│ result of applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.primary_key":
│ its value will be decided automatically based on the
│ result of applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.scale":
│ its value will be decided automatically based on the
│ result of applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.length":
│ its value will be decided automatically based on the
│ result of applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.nullable":
│ its value will be decided automatically based on the
│ result of applying this configuration.
╵
╷
│ Error: Value for unconfigurable attribute
│ 
│   with google_datastream_stream.default,
│   on example.tf line 115, in resource "google_datastream_stream" "default":
│  115:               ordinal_position = 1
│ 
│ Can't configure a value for
│ "source_config.0.sql_server_source_config.0.include_objects.0.schemas.0.tables.0.columns.0.ordinal_position":
│ its value will be decided automatically based on the
│ result of applying this configuration.
```

Reply:
Will mark this as resolved since 5e2cfb4d3fac327383f39715492092934361a8a8 reverts the removal of `output: true`

Reply:
New test output after compiling the provider again: 
```tf
TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set
│ in the CLI configuration:
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any released
│ version of the provider and applying changes may
│ cause the state to become incompatible with published
│ releases.
╵
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]

Terraform used the selected providers to generate the
following execution plan. Resource actions are
indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # google_datastream_stream.default will be created
  + resource "google_datastream_stream" "default" {
      + desired_state    = "NOT_STARTED"
      + display_name     = "sql-server-stream"
      + effective_labels = (known after apply)
      + id               = (known after apply)
      + location         = "us-east1"
      + name             = (known after apply)
      + project          = (sensitive value)
      + state            = (known after apply)
      + stream_id        = "sql-server-stream"
      + terraform_labels = (known after apply)

      + backfill_none {}

      + destination_config {
          + destination_connection_profile = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile"

          + gcs_destination_config {
              + file_rotation_interval = (known after apply)
              + file_rotation_mb       = (known after apply)
              + path                   = "stream-data"

              + json_file_format {
                  + compression        = "GZIP"
                  + schema_file_format = "NO_SCHEMA_FILE"
                }
            }
        }

      + source_config {
          + source_connection_profile = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile"

          + sql_server_source_config {
              + max_concurrent_backfill_tasks = (known after apply)
              + max_concurrent_cdc_tasks      = (known after apply)

              + include_objects {
                  + schemas {
                      + schema = (sensitive value)

                      + tables {
                          + table = (sensitive value)

                          + columns {
                              + column = "test"
                            }
                        }
                    }
                }
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

google_datastream_stream.default: Creating...
google_datastream_stream.default: Still creating... [10s elapsed]
google_datastream_stream.default: Creation complete after 11s [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set
│ in the CLI configuration:
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any released
│ version of the provider and applying changes may
│ cause the state to become incompatible with published
│ releases.
╵
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to generate the
following execution plan. Resource actions are
indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ length           = 0 -> 1
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides
│ are set in the CLI configuration:
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any
│ released version of the provider and applying
│ changes may cause the state to become
│ incompatible with published releases.
╵
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to
generate the following execution plan. Resource
actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ precision        = 0 -> 1
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides
│ are set in the CLI configuration:
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any
│ released version of the provider and applying
│ changes may cause the state to become
│ incompatible with published releases.
╵
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to
generate the following execution plan. Resource
actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ scale            = 0 -> 1
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides
│ are set in the CLI configuration:
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any
│ released version of the provider and applying
│ changes may cause the state to become
│ incompatible with published releases.
╵
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to
generate the following execution plan. Resource
actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ primary_key      = false -> true
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides
│ are set in the CLI configuration:
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any
│ released version of the provider and applying
│ changes may cause the state to become
│ incompatible with published releases.
╵
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to
generate the following execution plan. Resource
actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ nullable         = false -> true
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
nathan.bugden@Nathans-MBP telus-tpm % TF_CLI_CONFIG_FILE=/Users/nathan.bugden/terraform/magic-modules-dev-override.tfrc terraform apply --auto-approve
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides
│ are set in the CLI configuration:
│  - hashicorp/google-beta in /Users/nathan.bugden/go/bin
│  - hashicorp/google in /Users/nathan.bugden/go/bin
│ 
│ The behavior may therefore not match any
│ released version of the provider and applying
│ changes may cause the state to become
│ incompatible with published releases.
╵
google_datastream_connection_profile.destination_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-gcs-connection-profile]
google_datastream_private_connection.private_connection: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/privateConnections/sql-server-private-connection]
google_datastream_connection_profile.source_connection_profile: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/connectionProfiles/sql-server-connection-profile]
google_datastream_stream.default: Refreshing state... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]

Terraform used the selected providers to
generate the following execution plan. Resource
actions are indicated with the following
symbols:
  ~ update in-place

Terraform will perform the following actions:

  # google_datastream_stream.default will be updated in-place
  ~ resource "google_datastream_stream" "default" {
        id               = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        name             = "projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream"
        # (9 unchanged attributes hidden)

      ~ source_config {
            # (1 unchanged attribute hidden)

          ~ sql_server_source_config {
                # (2 unchanged attributes hidden)

              ~ include_objects {
                  ~ schemas {
                        # (1 unchanged attribute hidden)

                      ~ tables {
                            # (1 unchanged attribute hidden)

                          ~ columns {
                              ~ ordinal_position = 0 -> 1
                                # (6 unchanged attributes hidden)
                            }
                        }
                    }
                }
            }
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
google_datastream_stream.default: Modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream]
google_datastream_stream.default: Still modifying... [id=projects/prj-s-telus-tpm-data-fcfc/locations/us-east1/streams/sql-server-stream, 10s elapsed]
╷
│ Error: Error waiting for Updating Stream: 
│ 
│   with google_datastream_stream.default,
│   on example.tf line 91, in resource "google_datastream_stream" "default":
│   91: resource "google_datastream_stream" "default" {
│ 
╵
```

Reply:
Closed https://github.com/GoogleCloudPlatform/magic-modules/pull/10553#issuecomment-2099359433